### PR TITLE
Improve upper Hessenberg; document and test ilo/ihi

### LINKED ISSUE #2692: Further improvements of toUpperHessenberg
So that we don't miss the items detected in #2691 (these also apply to the related functions).

- Missing test-cases for non-default ilo, ihi.

- Documentation of ilo and ihi is unclear; basically the matrix is in supposed to be in BLT-form where the parts before and after ilo:ihi are already upper triangular and we only have to transform that sub-matrix.

In retrospect not exposing ilo and ihi had probably been a wiser choice, but removing them now seems more problematic.
(Removing the last column of V just seems to cause problems for no real gain.)

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #2692.
I believe this new documentation is more understandable - even if I don't see the obvious reason for exposing ilo/ihi.

### CODE REVIEW

File: Modelica/Math/package.mo [hunk 1]
```diff
@@ -10452,7 +10452,9 @@ The algorithm is taken from [1] and [2].
 <h4>Description</h4>
 <p>
 Function <strong>toUpperHessenberg</strong> computes a upper Hessenberg form <strong>H</strong> of a matrix <strong>A</strong> by orthogonal similarity transformation:  <strong>Q</strong>' * <strong>A</strong> * <strong>Q</strong> = <strong>H</strong>.
-With the optional inputs ilo and ihi, also partial transformation is possible. The function calls LAPACK function DGEHRD.
+The optional inputs <strong>ilo</strong> and <strong>ihi</strong> improve efficiency if the matrix is already partially converted to Hessenberg form; it is assumed
+that matrix <strong>A</strong> is already upper Hessenberg for rows and columns <strong>1:(ilo-1)</strong> and <strong>(ihi+1):size(A, 1)</strong>.
+The function calls LAPACK function DGEHRD.
```
Comment (beutlich) [inline Modelica/Math/package.mo]:
```suggestion
The function calls <a href=\"modelica://Modelica.Math.Matrices.LAPACK.dgehrd\">LAPACK.dgehrd</a>.
```

File: Modelica/Math/package.mo [hunk 2]
```diff
@@ -8000,9 +8000,9 @@ For details of the arguments, see documentation of dgbsv.
 
       input Real A[:, size(A, 1)];
       input Integer ilo=1
-        "Lowest index where the original matrix had been Hessenberg form";
+        "Lowest index where the original matrix have not been in upper triangular form";
```
Comment (beutlich) [inline Modelica/Math/package.mo]: have or has? I am not sure.

Comment (HansOlsson) [reply Modelica/Math/package.mo]: Should likely be singular; but on second thought the original matrix is unchanged so I believe it is simpler to use "is" leading to "where the original matrix is not in upper triangular form".

Comment (HansOlsson) [reply Modelica/Math/package.mo]: Shall I push the change to "is not in upper triangular form"?
Shall I also change this for ilo/ihi for dormhr and dorghr?
(As I understand they are all identical.)

Comment (beutlich) [reply Modelica/Math/package.mo]: Yes, please.

Comment (HansOlsson) [reply Modelica/Math/package.mo]: Done now.

**Pull request merged**